sup3r/postprocessing/file_handling.py

```diff
@@ -165,33 +256,44 @@ def get_lat_lon(low_res_lat_lon, shape):
             Last dimension has ordering (lat, lon)
         """
         logger.debug('Getting high resolution lat / lon grid')
-        s_enhance = shape[0] // low_res_lat_lon.shape[0]
-        old_points = np.zeros((np.product(low_res_lat_lon.shape[:-1]), 2),
-                              dtype=np.float32)
-        new_points = np.zeros((np.product(shape), 2), dtype=np.float32)
-        lats = low_res_lat_lon[..., 0].flatten()
-        lons = low_res_lat_lon[..., 1].flatten()
-
-        # This shifts the indices for the old points by the downsampling
-        # fraction so that we can calculate the centers of the new points with
-        # origin at zero. Obviously if the shapes are the same then there
-        # should be no shift
-        lat_shift = 1 - low_res_lat_lon.shape[0] / shape[0]
-        lon_shift = 1 - low_res_lat_lon.shape[1] / shape[1]
-
-        new_points[:, 0] = np.arange(new_points.shape[0]) // shape[1]
-        new_points[:, 1] = np.arange(new_points.shape[0]) % shape[1]
-        old_points[:, 0] = np.arange(old_points.shape[0])
-        old_points[:, 0] //= low_res_lat_lon.shape[1]
-        old_points[:, 0] *= s_enhance
-        old_points[:, 0] += lat_shift
-        old_points[:, 1] = np.arange(old_points.shape[0])
-        old_points[:, 1] %= low_res_lat_lon.shape[1]
-        old_points[:, 1] *= s_enhance
-        old_points[:, 1] += lon_shift
-        lats = RBFInterpolator(old_points, lats, neighbors=10)(new_points)
-        lons = RBFInterpolator(old_points, lons, neighbors=10)(new_points)
+
+        # ensure lons are between -180 and 180
+        low_res_lat_lon[..., 1] = (low_res_lat_lon[..., 1] + 180) % 360 - 180
+
+        # check if lons go through the 180 -> -180 boundary.
+        if not cls.is_increasing_lons(low_res_lat_lon):
+            low_res_lat_lon[..., 1] = (low_res_lat_lon[..., 1] + 360) % 360
+
+        # pad lat lon grid
+        padded_grid = cls.pad_lat_lon(low_res_lat_lon)
+        lats = padded_grid[..., 0].flatten()
+        lons = padded_grid[..., 1].flatten()
+
+        lr_y, lr_x = low_res_lat_lon.shape[:-1]
+        hr_y, hr_x = shape
+
+        # assume outer bounds of mesh (0, 10) w/ points on inside of that range
+        y = np.arange(0, 10, 10 / lr_y) + 5 / lr_y
+        x = np.arange(0, 10, 10 / lr_x) + 5 / lr_x
+
+        # add values due to padding
+        y = np.concatenate([[y[0] - 10 / lr_y], y, [y[-1] + 10 / lr_y]])
+        x = np.concatenate([[x[0] - 10 / lr_x], x, [x[-1] + 10 / lr_x]])
+
+        # remesh (0, 10) with high res spacing
+        new_y = np.arange(0, 10, 10 / hr_y) + 5 / hr_y
```

Comment:
I thought we still needed the RBFInterpolator for non-regular grids? Will linear interp of gridpoints be inaccurate for that funky WRF projection?

Reply:
No the griddata method works for general unstructured grids

Reply:
Gotcha, sounds good!